provisiondomainmanager_policy.tf

```diff
@@ -69,32 +69,41 @@ data "aws_iam_policy_document" "provisiondomainmanager_policy_acm_cognito_ec2_do
       "acm:DescribeCertificate",
       "acm:ListCertificates",
       "acm:RequestCertificate",
+      "cloudwatch:GetMetricData",
+      "cloudwatch:GetMetricStatistics",
+      "cloudwatch:ListMetrics",
       "cognito-idp:CreateUserPool",
       "cognito-idp:DescribeUserPool",
       "cognito-idp:DescribeUserPoolClient",
       "cognito-idp:DescribeUserPoolDomain",
       "cognito-idp:ListUserPools",
       "cognito-idp:TagResource",
+      "ec2:AssignPrivateIpAddresses",
```

Comment:
Do we create the EC2 instances with a tag?  Can we restrict this action to only those instances that are appropriately tagged?  Consider the same question for all actions in this block.

Reply:
Unfortunately, Private IP Addresses are assigned to network interfaces, and network interfaces are created in the background when generating resources like load balancers. I don't have the ability to tag these resources, and the IDs they are given don't follow a naming convention, so cannot restrict based on condition on resource.

Reply:
:disappointed: